src/compiler/checker.ts

```diff
@@ -23075,6 +23066,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return isArrayLikeType(type) || isTupleLikeType(type);
     }
 
+    function isAssignableToAvailableAnyIterable(type: Type): boolean {
+        if (!anyIterable) {
+            anyIterable = getGlobalIterableType(/*reportErrors*/ false) !== emptyGenericType ? createIterableType(anyType) : emptyGenericType;

```

Comment:
Should this be in a function like `getGlobalIteratorYieldResultType`? I think that's more usual for uncommon types that are obtained lazily from the globals.

Reply:
Of course, for future reusability purposes, I could create such a function anyway - if you prefer that.

Reply:
Foolishly I have misread this; `getGlobalIterableType` already _is_ one of these deferred-using functions, and was already used in the old code and is also called by `createIterableType`.

That being said, I'm not sure that caching `anyIterable` is worth it. `createIterableType` already re-calls `getGlobalIterableType(/*reportErrors*/ false)` so I don't think this is actually saving any time.

Reply:
I also wonder if an equivalent fix is to change `createTypeFromGenericGlobalType` to return `emptyGenericType` instead of `emptyObjectType`, which would net the same result here. But that's just a guess from skimming the code.

Reply:
(Making the change in the previous comment passes all tests on main, interesting.)

Reply:
pushed out a commit dropping `anyIterable` caching